### DISCUSSION TITLE
Pytest exit code 1 but all test seem to pass on Mac

### DISCUSSION THREAD

**Discussion by nojaf** [pytest-dev/pytest#13049] [Q&A]
Hello,

I'm losing my mind over a GitHub Action that runs pytest but keeps returning exit code 1.
In https://github.com/mlflow/mlflow-go/pull/95, I'm trying to isolate what is causing this but I can't seem to log anything useful.

The gist is that I'm running:

> uv run pytest -sv --log-cli-level=DEBUG -p no:warnings --confcutdir=. -k "not [file" .mlflow.repo/tests/tracking/test_model_registry.py

I do have a `conftest.py`, maybe something is up with that but I'm just pretty clueless at this point.

We also are not seeing this behavior on Windows or Linux on the main branch.

What can I enable to give more logging?

**Comment by The-Compiler** [comment]
Weird! What happens if you remove the `-p no:warnings`?

There's also `--debug` as pytest argument which writes a `pytestdebug.log` that you could e.g. `cat` afterwards to see what's happening internally in terms of hooks and such.

If that doesn't help, maybe try [hunter](https://python-hunter.readthedocs.io/en/latest/) to output tracing information about where sys.exit is called. I don't have something ready to trace `sys.exit` calls with it, maybe something like `PYTHONHUNTER="module='sys'"` or even `PYTHONHUNTER="stdlib=False"` is enough.

**Reply by nojaf** [reply]
Thanks, at first glance I don't see anything weird in https://github.com/mlflow/mlflow-go/actions/runs/12273643445/job/34244857471

I'll try hunter this afternoon.

**Reply by The-Compiler** [reply]
That looks even more weird to me:

- In its debug logs, pytest claims to exit with code 0
- Yet GitHub Actions then shows exit code 1
- ...but the `echo $?` shows 0 again (that might be a red herring though, I think every command in the YAML is in an independent shell)

You might want to try:

- Adding `; echo $?` directly to the command
- Maybe trying to run `pytest` in the virtualenv directly, without involving `uv run`? While it seems very unlikely that `uv run` actually changes the exit code, it would be good to have one fewer possible factor involved regardless.

**Reply by nojaf** [reply]
Wow, this is pretty wild but running `.venv/bin/pytest` does return the 0 exit code.
So this indeed might be a uv problem. Thanks for this suggestion!

**Reply by nojaf** [reply]
Okay, `uv` isn't the culprit after all.
I some cases, I see .`/Users/runner/work/_temp/d1c7eb06-b3f5-4afa-b2ab-f0952fae0cc2.sh: line 1:  4615 Segmentation fault: 11  .venv/bin/pytest --log-cli-level=DEBUG --confcutdir=. -k "not [file" -p no:warnings .mlflow.repo/tests/tracking/test_rest_tracking.py .mlflow.repo/tests/tracking/test_model_registry.py .mlflow.repo/tests/store/tracking/test_sqlalchemy_store.py .mlflow.repo/tests/store/model_registry/test_sqlalchemy_store.py`.

Where `Segmentation fault: 11` might mean something isn't disposed properly.
